results/user-testing-results.md

```diff
@@ -0,0 +1,138 @@
+
+#### Flexible command palette search/filter behavior
+
+Because several participants relied heavily on the command palette to complete tasks, the limit of its filtering arose several times. From the user end, the command palette feels like it searches only for direct matches within the command list for whatever the user types in. If a user knows exactly what they are looking for, they often have success. Participants were sometimes able to quickly and flawlessly locate a desired command. But when they were exploring JupyterLab’s command options, unable to remember the exact name of a command, made a typo, or were unfamiliar with a task many participants had to give up using the command palette in order to complete the task. Depending on the assistive tech they use, typos can also be nontrivial to detect for blind or visually impaired users. Having a fuzzy search or other typo/error support rather than only the direct filter would help the command palette continue to be the boon many participants wanted it to be.
+
+## For future tests
+
+Reflecting on the findings and gaps in this set of tests, the following are questions to revisit in the future.
+
+- How might a screen reader user complete common notebook tasks in JupyterLab?
+- What are JupyterLab’s strengths and weaknesses when it comes to authoring notebooks and other documents with assistive tech?
+- What more can we learn about navigating JupyterLab, especially with a variety of assistive tech? What are good examples of navigation that support disabled users?
+- What are the full range of notebook cell output types and what feedback to disabled users have on each type?
+- What is the real time collaboration experience in JupyterLab like for assistive tech users? This question requires JupyterLab collaboration modes to be more stable before it can be tested.
+
+As mentioned in the Limits section, the sample we were able to test with for these sessions was not as comprehensive of varied disabilities and assistive tech as we would like. Recruitment for these sessions was done within existing JupyterLab community spaces and was mostly self-selected; future tests would likely benefit from a more selective sample and outreach to disability communities where appropriate. This should be done with careful consideration as the history of people who make products interrupting disability communities and asking for labor is a fraught one.
```

Comment:
I wonder if moving this paragraph to the top and mentioning that certain features with huge accessibility implications (like RTC) were not tested. I think that would make it easier for the reader to understand why some of the things are in the list. Does that make sense?

Reply:
I can see an argument for it. Let me think on that one.

Reply:
Right now my brain outlines this section as so:

- section heading: For future tests
- list of questions to explore further
- paragraph about the need to improve the selection sample
- paragraph about the need to use other user research methodologies

Looking at it this way, I take back my suggestion about moving the paragraph in the third bullet point to the top of the section.

However, because there are some things in the list that the preceding parts of the document did not touch upon, or perhaps only vaguely touched upon, it might make sense to insert some more wording between the section heading and the list, so as not to interrupt the reader's flow.

Let me go through the list of questions, one by one, trying to give you a window into my inner reader voice, showing which questions felt like they were in conversation with previous parts of the document versus ones which felt disconnected:

- Screen readers. My inner voice: this seems in conversation with the "Limits" section, specifically the part about not having any screen reader participants
- Authoring. Me: Unclear. disconnected? but maybe in conversation with the "notebook" section, since it seems that much of the findings were about users consuming a notebook rather than producing one
- Navigating. Me: this is definitely in conversation with multiple parts of the document, including the navigation findings section, as well as the suggestions for improvement
- Cell output types and feedback. Me: vague. in conversation with... the part about cell status and the drastic experience difference between non-sighted versus sighted users?
- RTC. Me: disconnected.

Hope that helps!

Reply:
I appreciate the explanation. If you are looking for relation between this section and the rest of the document, I would say it's mostly not there. 

I'll add a suggestion with some changes to the intro to see if that clarifies. I have to add it in another comment though to capture the right lines to change. Thanks!